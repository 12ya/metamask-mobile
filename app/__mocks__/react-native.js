import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
