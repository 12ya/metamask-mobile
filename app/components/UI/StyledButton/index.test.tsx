import React from 'react';
import { render } from '@testing-library/react-native';
import StyledButtonIOS from './index.ios';
import StyledButtonAndroid from './index.android';

describe('StyledButton', () => {
  const buttonTypes = ['normal', 'confirm', 'orange', 'cancel'];

  buttonTypes.forEach((type) => {
    it(`should render correctly on iOS the button with type ${type}`, () => {
      const { toJSON } = render(<StyledButtonIOS type={type} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  buttonTypes.forEach((type) => {
    it(`should render correctly on Android the button with type ${type}`, () => {
      const { toJSON } = render(<StyledButtonAndroid type={type} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
