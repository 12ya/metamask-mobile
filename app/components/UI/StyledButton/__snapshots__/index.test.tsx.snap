// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton should render correctly on Android the button with type cancel 1`] = `
<TouchableNativeFeedback
  accessibilityRole="button"
  accessible={true}
>
  <View
    style={
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#6a737d",
          "borderWidth": 1,
        },
        null,
        undefined,
      ]
    }
  />
</TouchableNativeFeedback>
`;

exports[`StyledButton should render correctly on Android the button with type confirm 1`] = `
<TouchableNativeFeedback
  accessibilityRole="button"
  accessible={true}
>
  <View
    style={
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#0376c9",
          "minHeight": 50,
        },
        null,
        undefined,
      ]
    }
  />
</TouchableNativeFeedback>
`;

exports[`StyledButton should render correctly on Android the button with type normal 1`] = `
<TouchableNativeFeedback
  accessibilityRole="button"
  accessible={true}
>
  <View
    style={
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#0376c9",
          "borderWidth": 1,
        },
        null,
        undefined,
      ]
    }
  />
</TouchableNativeFeedback>
`;

exports[`StyledButton should render correctly on Android the button with type orange 1`] = `
<TouchableNativeFeedback
  accessibilityRole="button"
  accessible={true}
>
  <View
    style={
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "borderColor": "#bf5208",
          "borderWidth": 1,
        },
        null,
        undefined,
      ]
    }
  />
</TouchableNativeFeedback>
`;

exports[`StyledButton should render correctly on iOS the button with type cancel 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  accessible={true}
  activeOpacity={0.2}
  style={
    [
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#6a737d",
          "borderWidth": 1,
        },
        undefined,
      ],
      null,
    ]
  }
/>
`;

exports[`StyledButton should render correctly on iOS the button with type confirm 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  accessible={true}
  activeOpacity={0.2}
  style={
    [
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#0376c9",
          "minHeight": 50,
        },
        undefined,
      ],
      null,
    ]
  }
/>
`;

exports[`StyledButton should render correctly on iOS the button with type normal 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  accessible={true}
  activeOpacity={0.2}
  style={
    [
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#0376c9",
          "borderWidth": 1,
        },
        undefined,
      ],
      null,
    ]
  }
/>
`;

exports[`StyledButton should render correctly on iOS the button with type orange 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  accessible={true}
  activeOpacity={0.2}
  style={
    [
      [
        {
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 15,
        },
        {
          "borderColor": "#bf5208",
          "borderWidth": 1,
        },
        undefined,
      ],
      null,
    ]
  }
/>
`;
